src/subscription_manager/i18n.py

```diff
@@ -51,10 +54,100 @@ def configure_gettext():
+
+
+class Locale(object):
+    """
+    Class used for changing languages on the fly
+    """
+
+    languages = {}
+    converter = {}
+
+    @classmethod
+    def set_gettext_method(cls, language):
+        """
+        Set gettext method and locale for specified language. This method is
+        intended for changing language on the fly, because rhsm service can
+        be used by many users with different language preferences at the
+        same time.
+        :param language: String representing locale
+                (e.g. de, de_DE, de_DE.utf-8, de_DE.UTF-8)
+        """
+        global ugettext, ungettext
+        lang = None
+
+        if language != '':
+            # When we already found some working alternative for language code, then use it
+            if language in cls.converter:
+                language = cls.converter[language]
+            if language not in cls.languages.keys():
+                # Try to find given language
+                try:
+                    lang = gettext.translation(APP, DIR, languages=[language])
+                except IOError as err:
+                    log.info('Could not import locale for %s: %s' % (language, err))
+                    # When original language was not found, then we will try another
+                    # alternatives.
+                    orig_language = None
+                    # For similar case: 'de'
+                    if '_' not in language:
+                        orig_language = language
+                        language += '_' + language.upper()
+                    # For similar cases: 'de_AT' (Austria), 'de_LU' (Luxembourg)
+                    elif language[0:2] != language[3:5]:
+                        orig_language = language
+                        language = language[0:2] + '_' + language[0:2].upper() + language[5:]
+                    if orig_language:
+                        try:
+                            lang = gettext.translation(APP, DIR, languages=[language])
+                        except IOError as err:
+                            log.info('Could not import locale either for %s: %s' % (language, err))
+                        else:
+                            log.debug('Using new locale for language: %s' % language)
+                            cls.languages[language] = lang
+                            cls.converter[orig_language] = language
+                else:
+                    log.debug('Using new locale for language: %s' % language)
+                    cls.languages[language] = lang
+            else:
+                log.debug('Reusing locale for language: %s' % language)
+                lang = cls.languages[language]
+
+        if lang is not None:
+
+            # Set locale, because rhsm.connection use it
+            if '_' not in language:
+                language += '_' + language.upper()
+            if language.upper().endswith('.UTF-8'):
+                locale.setlocale(locale.LC_ALL, language)
```

Comment:
Same concern as above question regarding race conditions... We might have to do some hacking to get locale to be per-thread for rhsm connection if so...

Reply:
Yes, this could be issue. Overall I don't like the idea of using environment variable for storing current language. I will try to find different solution.